File: easy/reverse_integer/reverse_integer.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func reverseString(str string) string {
	charArr := []rune(str)

	i, j := 0, len(charArr)-1
	for i < j {
		charArr[i], charArr[j] = charArr[j], charArr[i]
		i++
		j--
	}

	return string(charArr)
}

func reverse(x int) int {
	if x > math.MaxInt32 || x < math.MinInt32 {
		return 0
	}
	negative := false
	if x < 0 {
		negative = true
		x = x * -1
	}

	str := strconv.Itoa(x)
	reverseStr := reverseString(str)
	reverseInt, err := strconv.ParseInt(reverseStr, 10, 32)
	if err != nil {
		return 0
	}

	if reverseInt > math.MaxInt32 || reverseInt < math.MinInt32 {
		return 0
	}
	if negative {
		reverseInt = reverseInt * -1
	}
	return int(reverseInt)
}

func main() {
	fmt.Println("Solution 1 : ", reverse(123))
	fmt.Println("Solution 2 : ", reverse(-123))
	fmt.Println("Solution 3 : ", reverse(120))
	fmt.Println(math.MaxInt32)
}

File: easy/array/reverse_integer/reverse_integer.py
import numpy


class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        INTMAX32 = 2147483647
        if abs(x) > INTMAX32:
            return 0

        negative = False
        if x < 0:
            negative = True
            x = abs(x)

        str_x = str(x)
        reverse_str = str_x[::-1]
        reverse_int = int(reverse_str)

        if reverse_int > INTMAX32:
            return 0
        if negative:
            reverse_int = reverse_int * -1

        return reverse_int


s = Solution()
print("Solution 1 : ", s.reverse(123))
print("Solution 2 : ", s.reverse(-123))
print("Solution 3 : ", s.reverse(120))
